src/main/kotlin/com/coder/jetbrains/services/CoderPortForwardService.kt

```diff
@@ -47,13 +56,54 @@ class CoderPortForwardService(
         poller?.cancel()
     }
 
+    class InvalidJsonTypeException(message: String) : Exception(message)
+
     private fun start() {
-        logger.info("starting port scanner")
+        val devcontainerFile = CoderBackendSettings.getDevcontainerFile()
+        if (devcontainerFile.exists()) {
+            try {
+                val json = devcontainerFile.readText()
+                val obj = JSONObject(json)
```

Comment:
I need to double check but as far as I know IntelliJ framework already provides support for JSON through kotlinx.serialization library. Can we use that instead?

Reply:
Switched to `kotlinx.serialization`